botocore/awsrequest.py

```diff
@@ -123,7 +123,9 @@ def _send_request(self, method, url, body, headers):
 
     def _send_output(self, message_body=None):
         self._buffer.extend((b"", b""))
-        msg = b"\r\n".join(self._buffer)
+        msg = b"\r\n".join(
+            b.encode('utf-8') if isinstance(b, six.text_type) else b
```

Comment:
I'd prefer if it was not all in-line as it took me a little longer to understand. Code looks fine otherwise.